waveform_editor/configuration.py

```diff
@@ -69,6 +79,16 @@ def add_waveform(self, waveform, path):
         group = self.traverse(path)
         group.waveforms[waveform.name] = waveform
         self.waveform_map[waveform.name] = group
+        if isinstance(waveform, DerivedWaveform):
+            try:
+                self.dependency_graph.add_node(
+                    waveform.name, waveform.dependent_waveforms
+                )
```

Comment:
Can you do this check before adding it to `group.waveforms` and `self.waveform_map`? Then you can remove the try/except as you wouldn't need to execute the `del` statements.